relay_test.go

```diff
@@ -1584,18 +1586,46 @@ func TestRelayModifyArg2(t *testing.T) {
 			arg3: []byte{},
 		},
 		{
-			msg: "small payloads",
+			msg: "1kB payloads",
```

Comment:
couple more corner cases just to verify we're not missing other issues: large header values that causes arg2 to end right near the boundary of the first frame (ideally both a few bytes before, 2 bytes before, 1 byte before, and on the exact boundary)

Ideally we'd test that with all sort of arg3 values, so i think we may want to do all combinations of:
 * no arg2, small arg2, large arg2 (3 or 4 cases that are close/on the boundary)
 * no arg3, small arg3,  16kb arg3, 32kb arg3, 64kb arg3, 128kb arg3, 1mb arg3

(I can help with the framework part of the test in a separate PR)

Reply:
added TODO for following up afterward